src/Domain/DonationRepository.php

```diff
@@ -544,7 +544,15 @@ public function abandonOldCancelled(): int
         return count($donations);
     }
 
-    public function deriveFees(Donation $donation, ?string $cardBrand = null, ?string $cardCountry = null): void
+    /**
+     * Updates a donation to set the appropriate fees. If card details are null then we assume for now that a card with
+     * the lowest possible fees will be used, and this should be called again with the details of the selected card
+     * when confirming the payment.
+     *
+     * @psalm-param value-of<Calculator::STRIPE_CARD_BRANDS>|null $cardBrand
+     * @param string|null $cardCountry ISO two letter uppercase code
+     */
+    public function deriveFees(Donation $donation, ?string $cardBrand, ?string $cardCountry): void
```

Comment:
I thought it worth removing the default nulls so that we're forced to look at the nulls and think about them when we call this.